snuba/datasets/profiles_processor.py

```diff
@@ -0,0 +1,38 @@
+from datetime import datetime
+from typing import Any, Mapping, Optional
+
+from snuba.consumers.types import KafkaMessageMetadata
+from snuba.processor import InsertBatch, MessageProcessor, ProcessedMessage
+
+
+class ProfilesProcessor(MessageProcessor):
+    def process_message(
+        self, message: Mapping[str, Any], metadata: KafkaMessageMetadata
+    ) -> Optional[ProcessedMessage]:
+        try:
+            processed = {
+                "organization_id": message["organization_id"],
+                "project_id": message["project_id"],
+                "transaction_id": message["transaction_id"],
+                "received": datetime.fromisoformat(message["received"]),
+                "profile": message["profile"],
+                "symbols": message["symbols"],
+                "android_api_level": message.get("android_api_level"),
+                "device_classification": message["device_classification"],
+                "device_locale": message["device_locale"],
+                "device_manufacturer": message["device_manufacturer"],
+                "device_model": message["device_model"],
+                "device_os_build_number": message.get("device_os_build_number"),
+                "device_os_name": message["device_os_name"],
+                "device_os_version": message["device_os_version"],
+                "duration_ns": message["duration_ns"],
+                "environment": message.get("environment"),
+                "platform": message["platform"],
+                "trace_id": message["trace_id"],
+                "version_name": message["version_name"],
+                "version_code": message["version_code"],
+                "retention_days": 30,
+            }
+        except KeyError:
+            return None
```

Comment:
I would add a metric or a log here so you can debug this in case it breaks. You can look at the transactions processor for an example of how to set up metrics in a processor.